trainee_template/ablation.amlg

```diff
@@ -0,0 +1,356 @@
+
+            ;internal only
+            react_kwargs (assoc)
+        )
+
+        (if (not use_case_weights)
+            (assign (assoc weight_feature ".none"))
+        )
+        
+        (assign (assoc
+            react_kwargs
+                (assoc
+                    context_features context_features
+                    use_case_weights use_case_weights
+                    weight_feature weight_feature
+                    details (assoc influential_cases (true) )
+                )
+        ))
+        
+        (if 
+            ;if context_values are not null, we are reacting to a new case
+            (not (= context_values (null)))
+
+            (accum "react_kwargs" (assoc context_values context_values) )
+            ;elif case_id is not null, we are reacting to an existing case
+            (not (= case_id (null)))
+            (accum "react_kwargs" (assoc
+                case_indices (retrieve_from_entity case_id (list ".session" ".session_training_index") )
+                preserve_feature_values context_features
+                leave_case_out (true)
+            ))
+        )
+
+        ;compute the entropy of the influence weights as retrieved by react
+        (entropy (map
+            (lambda (get (current_value) ".influence_weight") )
+            (get (call React react_kwargs) "influential_cases")
+        ))
+    )
+
+    ;used to compute and store the influence weight entropies for all cases contained in the Trainee in #Analyze
+    ; context_features: list of context features to use when determining influential cases.
+    ; label_name: optional name of the feature to store influence weight entropies in.
+    #ComputeAndStoreInfluenceWeightEntropies
+    (let 
+        (assoc
+            influence_weight_entropy_map
+                ||(map
+                    (lambda
+                        (call ComputeInfluenceWeightEntropy (assoc
+                            case_id (current_index 1)
+                            context_features context_features
+                            use_case_weights use_case_weights
+                            weight_feature weight_feature
+                        ))
+                    )
+                    (zip (call AllCases))
+                )
+        )
+
+        (call StoreCaseValues (assoc
+            case_values_map influence_weight_entropy_map
+            label_name
+                (if (= (null) label_name)
+                    internalLabelInfluenceWeightEntropy
+                    label_name
+                )
+        ))
+
+        (accum_to_entities (assoc revision 1))
+        (assign_to_entities (assoc hasInfluenceWeightEntropies (true)))
+    )
+
+    ;initialize parameters and internal features related to auto ablation
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #InitializeAutoAblate
+    (declare
+        (assoc weight_feature autoAblateWeightFeature)
+
+        (seq
```

Comment:
don't need this `seq` , its statements are already in a declare statement where everything after the (assoc) above is executed in sequence